lib/coll.gi

```diff
@@ -3052,6 +3053,31 @@ end);
 InstallMethod( CanComputeIsSubset,"default: no, unless identical",
   [IsObject,IsObject],IsIdenticalObj);
 
+# avoid immediate methods triggering multiple type changes once the Size of
+# an object is known.
+InstallOtherMethod(SetSize,true,[IsObject and IsAttributeStoringRep,IsObject],
+  100, # override system setter
+function(obj,sz)
+local filt;
+  if HasSize(obj) and Size(obj)<>sz then
+    if AssertionLevel()>2 then
+      # Make this an ordinary error to enter break loop so that one can
+      # investigate call order for debugging
```

Comment:
I don't understand this comment. What is a "non-ordinary error"? Perhaps you are referring to `Assert`? But that enters a break loop for me just fine, doesn't it?

Reply:
You had suggested `ErrorNoReturn` in an earlier remark on this function. For debugging (and understanding how code worked in the past despite this issue) it seemed better to not forbid `return;`.  (Yes, it could mislead a user, but this is really an error that should never come up for a user. (I'm happy to replace by `Assert`, if that is preferred here.